src/server/pfs/fuse/server.go

```diff
@@ -188,27 +188,41 @@ func (mm *MountManager) ListByRepos() (ListRepoResponse, error) {
 	}
 	for _, repo := range repos {
 		rr := RepoResponse{Name: repo.Repo.Name, Branches: map[string]BranchResponse{}}
-		bs, err := mm.Client.ListBranch(repo.Repo.Name)
-		if err != nil {
-			return lr, err
+		readAccess := true
+		if repo.AuthInfo != nil {
+			readAccess = hasRepoRead(repo.AuthInfo.Permissions)
+			rr.Authorization = "none"
 		}
-		for _, branch := range bs {
-			br := BranchResponse{Name: branch.Branch.Name}
-			k := MountKey{
-				Repo:   repo.Repo.Name,
-				Branch: branch.Branch.Name,
-				Commit: "",
+		if readAccess {
+			bs, err := mm.Client.ListBranch(repo.Repo.Name)
+			if err != nil {
+				return lr, err
 			}
-			// Add all mounts associated with a repo/branch
-			for _, msm := range mm.States {
-				if msm.MountKey == k && msm.State != "unmounted" {
-					br.Mount = append(br.Mount, msm.MountState)
+			for _, branch := range bs {
+				br := BranchResponse{Name: branch.Branch.Name}
+				k := MountKey{
+					Repo:   repo.Repo.Name,
+					Branch: branch.Branch.Name,
+					Commit: "",
 				}
+				// Add all mounts associated with a repo/branch
+				for _, msm := range mm.States {
+					if msm.MountKey == k && msm.State != "unmounted" {
+						br.Mount = append(br.Mount, msm.MountState)
+					}
+				}
+				if br.Mount == nil {
+					br.Mount = append(br.Mount, MountState{State: "unmounted"})
+				}
+				rr.Branches[branch.Branch.Name] = br
 			}
-			if br.Mount == nil {
-				br.Mount = append(br.Mount, MountState{State: "unmounted"})
+			if repo.AuthInfo == nil {
+				rr.Authorization = "off"
```

Comment:
This is the case for workspaces where auth is not enabled correct?

Reply:
off- auth disabled
none- auth enabled and user has no permissions